internal/convert/registryv1.go

```diff
@@ -98,127 +229,101 @@ func Convert(in RegistryV1, installNamespace string, targetNamespaces []string)
+	//   2. There's an extra RoleBinding for the above mentioned role.
+	//        - Every SA mentioned in the OperatorCondition.spec.serviceAccounts is a subject for this role binding: https://github.com/operator-framework/operator-lifecycle-manager/blob/9ced412f3e263b8827680dc0ad3477327cd9a508/pkg/controller/operators/operatorcondition_controller.go#L171-L177
+	//   3. strategySpec.permissions are _also_ given a clusterrole/clusterrole binding.
+	//  		- (for AllNamespaces mode only?)
+	//			- (where does the extra namespaces get/list/watch rule come from?)
+
 	roles := []rbacv1.Role{}
 	roleBindings := []rbacv1.RoleBinding{}
 	clusterRoles := []rbacv1.ClusterRole{}
 	clusterRoleBindings := []rbacv1.ClusterRoleBinding{}
 
-	for _, ns := range targetNamespaces {
-		if ns == "" {
-			continue
+	permissions := in.CSV.Spec.InstallStrategy.StrategySpec.Permissions
+	clusterPermissions := in.CSV.Spec.InstallStrategy.StrategySpec.ClusterPermissions
+	allPermissions := append(permissions, clusterPermissions...)
+
+	// Create all the service accounts
+	for _, permission := range allPermissions {
+		saName := saNameOrDefault(permission.ServiceAccountName)
+		if _, ok := serviceAccounts[saName]; !ok {
+			serviceAccounts[saName] = newServiceAccount(installNamespace, saName)
 		}
-		for _, permission := range in.CSV.Spec.InstallStrategy.StrategySpec.Permissions {
-			if _, ok := serviceAccounts[permission.ServiceAccountName]; !ok {
-				serviceAccounts[permission.ServiceAccountName] = corev1.ServiceAccount{
-					TypeMeta: metav1.TypeMeta{
-						Kind:       "ServiceAccount",
-						APIVersion: corev1.SchemeGroupVersion.String(),
-					},
-					ObjectMeta: metav1.ObjectMeta{
-						Namespace: installNamespace,
-						Name:      permission.ServiceAccountName,
-					},
-				}
-			}
-			name := generateName(fmt.Sprintf("%s-%s", in.CSV.GetName(), permission.ServiceAccountName), []interface{}{in.CSV.GetName(), permission})
-			roles = append(roles, rbacv1.Role{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       "Role",
-					APIVersion: rbacv1.SchemeGroupVersion.String(),
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: ns,
-					Name:      name,
-				},
-				Rules: permission.Rules,
-			})
-			roleBindings = append(roleBindings, rbacv1.RoleBinding{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       "RoleBinding",
-					APIVersion: rbacv1.SchemeGroupVersion.String(),
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Namespace: ns,
-					Name:      name,
-				},
-				Subjects: []rbacv1.Subject{
-					{
-						Kind:      "ServiceAccount",
-						Name:      permission.ServiceAccountName,
-						Namespace: installNamespace,
-					},
-				},
-				RoleRef: rbacv1.RoleRef{
-					APIGroup: rbacv1.GroupName,
-					Kind:     "Role",
-					Name:     name,
-				},
+	}
+
+	//oc := &v2.OperatorCondition{
```

Comment:
nit: should we remove these comments?

Reply:
This is all the stuff related to supporting OperatorCondition generation. There's a reasonable chance we'll need this, but I agree that keeping all these comments isn't the best.

I'll add another commit that drops the comments. That way these comments will be in the history so we can revive them if necessary.